### PR TITLE
remove trailing slashes from api urls and return empty list when there are no users

### PR DESCRIPTION
**Pull request by shannoncolavita**
*Issue #, if available:*

*Description of changes:*
1. Removed trailing slashes from api urls as this results in urls with double slashes, sometimes resulting in CORS issues.
2. Return empty list when there are no users as this results in API Gateway error as null was being returned 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Lab1/scripts/deployment.sh [hunk 1]
```diff
@@ -72,7 +72,9 @@ if [[ $client -eq 1 ]]; then
   APP_SITE_BUCKET=$(aws cloudformation describe-stacks --stack-name "$stackname" --query "Stacks[0].Outputs[?OutputKey=='AppBucket'].OutputValue" --output text)
   APP_SITE_URL=$(aws cloudformation describe-stacks --stack-name "$stackname" --query "Stacks[0].Outputs[?OutputKey=='ApplicationSite'].OutputValue" --output text)
   APP_APIGATEWAYURL=$(aws cloudformation describe-stacks --stack-name "$stackname" --query "Stacks[0].Outputs[?OutputKey=='APIGatewayURL'].OutputValue" --output text)
-
+  
+  # remove trailing slash from url
+  APP_APIGATEWAYURL=${APP_APIGATEWAYURL%/}
```
Comment (ujwalbukka) [inline Lab1/scripts/deployment.sh:77]: Actually in the Cloudformation template, output value of these api gateway url output parameters have a trailing slash appended. I think we should remove the trailing slash for those output parameters in the cloudformation templates. You need to update server/shared-template.yaml and server/tenant-template.yaml files in each lab.

File: Lab2/server/TenantManagementService/user-management.py [hunk 2]
```diff
@@ -73,8 +73,9 @@ def get_users(event, context):
             user_info.user_name = user["Username"]
             users.append(user_info)                    
     
-        return utils.generate_response(users)
-   
+    # return an empty list when there are no users otherwise will result in API Gateway error
+    return utils.generate_response(users)
+
```
Comment (ujwalbukka) [inline Lab2/server/TenantManagementService/user-management.py:78]: Good catch. Thanks. Will be waiting on above fix to merge everything.

**Review by ujwalbukka — CHANGES_REQUESTED** [review]
Please review the comments and do the needful. Thanks.

**Comment by shannoncolavita** [comment]
Please review again when you get a chance, have removed the trailing slashes from the sam templates

**Comment by shannoncolavita** [comment]
@ujwalbukka Please review again when you get a chance, have removed the trailing slashes from the sam templates and reverted the changes to the deployment.sh files, thanks!

**Review by ujwalbukka — APPROVED** [review]

**Pull request merged**